### PR TITLE
Associations between User, Events and Event Attendance

### PR DESCRIPTION
**Pull request by mcastorena0316**
In this branch we create: 

-  Models: User, Event, and EventAttendance.
- The associations between the models using has_many and belongs_to.
- Controllers: Users and Events with their methods.
- Views for Login, Signup, Display User information with his/her created events, and his/her upcoming and previous events.
- Testing for User, Event and Integration.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    if logged_in?
+      @event = current_user.events.build
+      @feed_items = current_user.feed.paginate(page: params[:page])
+      @prev_events = []
+      @upcoming_events = []
+      @user.event_attendance.each do |x|
+        y = Event.find_by(id: x.attended_event)
+        @upcoming_events << y if y.date >= DateTime.now
+        @prev_events << y if y.date < DateTime.now
+      end
```
Comment (maelfosso) [inline app/controllers/users_controller.rb]: 1. It's the user's show page. So you should find the user from his id not only the `current_user`

2. In a few lines, 
```ruby
@prev_events = @user.events.prev_events
@upcoming_events = @user.events.upcoming_events
```

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      flash[:success] = "Welcome to Private Events!"
+      redirect_to @user
+    else
+      render 'new'
+      end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    if logged_in?
```
Comment (maelfosso) [inline app/controllers/users_controller.rb:20]: I suggest you set `before_action` with a redirect if it's not connected

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @mcastorena0316 @Luzaks 

Your code is almost good. Just a few steps to make fully good.

1. According to what is described [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png), the user's show page displays events the user has created separated in past and upcoming
But from this screenshot, the events are not separated
Screenshot from 2020-01-29 04-29-17

2. Also, it's specified [here] that the user should log in using **email** instead of `name` as you do.

3. In `event_attendances`, please add indexes on referential fields.

4. Also, read the comments below.

Kindly make that changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by wathigo — APPROVED** [review]
## Approved :heavy_check_mark:  
### Great work implementing the suggested changes :clap: 
**You may move on.**

**Good luck :four_leaf_clover: 
Simon Wathigo**

**Pull request merged**